fml/native_library.cc

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/native_library.h"
+
+#include <dlfcn.h>
+#include <fcntl.h>
+
+namespace fml {
+
+NativeLibrary::NativeLibrary(const char* path) {
+  ::dlerror();
+  handle_ = ::dlopen(path, RTLD_NOW);
+  const char* error = ::dlerror();
+  if (error != nullptr) {
+    handle_ = nullptr;
```

Comment:
I thought `dlopen` always returned `nullptr` in case of failure, in which case, maybe make the conditional above a null check on handle_ then fetch the error in here?

Reply:
Done.